assets/js/frontend/course-video/video-blocks-manager.js

```diff
@@ -0,0 +1,154 @@
+/**
+ * The Course Video Progression settings.
+ */
+const {
+	courseVideoRequired,
+	courseVideoAutoComplete,
+	courseVideoAutoPause,
+} = window.sensei.courseVideoSettings;
+
+/**
+ * Map of videos.
+ */
+const videos = {};
+
+/**
+ * Registers the videos to manage.
+ * @param {Object}   video
+ * @param {Function} video.pauseVideo              The function that pauses the video when invoked.
+ * @param {Function} video.registerVideoEndHandler Accepts a callback that is invoked when the video ends.
+ * @param {string}   video.url                     The source url of the video. Used as an id.
+ */
+export const registerVideo = ( {
+	pauseVideo = () => {},
+	registerVideoEndHandler = () => {},
+	url = '',
+} ) => {
+	const blocksStore = window.sensei?.store?.blocks;
+	if ( courseVideoRequired ) {
```

Comment:
Although I understand that this is a matter of personal preferance, I find dense code blocks like these hard to read. Maybe adding a couple of blank lines would improve things?

Reply:
Updated the whole logic to use hooks now. Let me know if it's better now.

Reply:
My initial thought was to use a single filter and pass `registerVideo` in it. Do you think that this is possible?